x/concentrated-liquidity/swapstrategy/one_for_zero.go

```diff
@@ -87,6 +87,23 @@ func (s oneForZeroStrategy) ComputeSwapWithinBucketOutGivenIn(sqrtPriceCurrent,
 	// Note that spread reward is always charged on the amount in.
 	spreadRewardChargeTotal := computeSpreadRewardChargePerSwapStepOutGivenIn(hasReachedTarget, amountOneIn, amountOneInRemaining, s.spreadFactor)
 
+	fmt.Println("amountOneIn", amountOneIn)
+	fmt.Println("amountOneInRemaining", amountOneInRemaining)
+	fmt.Println("sqrtPriceCurrent", sqrtPriceCurrent)
+	fmt.Println("sqrtPriceNext", sqrtPriceNext)
+
+	// This covers an edge case where due to the lack of precision, the difference between the current sqrt price and the next sqrt price is so small that
+	// it ends up being rounded down to zero. This leads to an infinite loop in the swap algorithm. From knowing that this is a case where !hasReachedTarget,
+	//(that is the swap stops within a bucket), we charge the full amount remaining in to the user and infer the amount out from the sqrt price truncated
+	// in favor of the pool.
+	if !hasReachedTarget && sqrtPriceCurrent.Equal(sqrtPriceNext) && amountOneIn.IsZero() && !amountOneInRemaining.IsZero() {
+		amountOneIn = amountOneInRemaining
+		// Subtract 1 ULP so that the amount out is rounded in favor of the pool.
+		// TODO: figure out precision difference in the denominations of two tokens,
+		// leading to incorrect spot price in out core logic.
+		amountZeroOut = sqrtPriceCurrent.Sub(oneULP).PowerMut(2)
```

Comment:
I'm referring to the issue described [here](https://github.com/osmosis-labs/osmosis/blob/83a14999fa5cc813f9dbc2f763b3917044ef4c5a/x/concentrated-liquidity/README.md?plain=1#L1628-L1638)